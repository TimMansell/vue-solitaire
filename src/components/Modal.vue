<template>
  <modal
    ref="modal"
    name="modal"
    classes="modal"
    height="auto"
    :scrollable="true"
    :reset="true"
    :adaptive="true"
    transition="scale"
  >
    <div class="modal__content" data-test="modal-content">
      <button class="modal__close" @click="closeModal" data-test="close-modal">
        Close
      </button>
      <slot />
    </div>
  </modal>
</template>

<script>
export default {
  name: 'Controls',
  props: {
    isVisible: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    closeModal() {
      this.$modal.hide('modal');
    },
  },
};
</script>

<style lang="scss">
.modal {
  &__close {
    position: absolute;
    right: 2rem;
  }

  &__content {
    padding: 1rem;
    background-color: #fff;
    text-align: left;
    border-radius: 3px;
    box-shadow: 0 0 20px -2px rgba(27, 33, 58, 0.4);
    margin: 1rem;
    left: 0;
    right: 0;
  }
}
</style>
