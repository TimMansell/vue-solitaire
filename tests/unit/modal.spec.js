import { createLocalVue, shallowMount } from '@vue/test-utils';
import VModal from 'vue-js-modal';
import Modal from '@/components/Modal.vue';

const localVue = createLocalVue();
localVue.use(VModal);

describe('Modal.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Modal, { localVue });

    expect(wrapper).toMatchSnapshot();
  });
});
