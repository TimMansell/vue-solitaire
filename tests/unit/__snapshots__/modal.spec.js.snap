// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal.vue matches snapshot 1`] = `
<modal-stub name="modal" delay="0" adaptive="true" scrollable="true" reset="true" overlaytransition="overlay-fade" transition="scale" clicktoclose="true" classes="modal" minwidth="0" minheight="0" maxwidth="9007199254740991" maxheight="9007199254740991" width="600" height="auto" pivotx="0.5" pivoty="0.5">
  <div data-test="modal-content" class="modal__content">
    <button-stub alt="true" data-test="close-modal" class="modal__close">
      Close
    </button-stub>
  </div>
</modal-stub>
`;
